main/lsp/LSPMessage.h

```diff
@@ -62,8 +62,9 @@ class LSPMessage final {
      * this one. */
     std::vector<std::unique_ptr<Timer>> timers;
 
-    /** Request counter. */
-    int counter;
+    /** Internal request id. Used to track unique messages internally. Used instead of id() since notification messages
+     * lack IDs. */
+    std::optional<int> internalId = std::nullopt;
```

Comment:
when would it be not set? can we make it always set? Otherwise, could we document when it's not set

Reply:
It's not set until it's passed to the preprocessor thread, which sets it.

It could default to 0. I made it `optional` so I can ENFORCE when it's not set.